From c9a2b220825754d374e0f7595006161644074b7a Mon Sep 17 00:00:00 2001
From: "qiang.cheng" <qiang.cheng@amlogic.com>
Date: Tue, 16 May 2023 09:40:06 +0800
Subject: [PATCH 1/8] yocto: add new judgment for feature arm-isp [1/1]

PD#SWPL-123911

Problem:
isp ko not install

Solution:
add new judgment for feature arm-isp

Verify:
local g12b

Signed-off-by: qiang.cheng <qiang.cheng@amlogic.com>
Change-Id: Ibbc977b7b45ed1f5355ea4eb042c781608166296
---
 recipes-kernel/isp/isp_5.15.bb | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/recipes-kernel/isp/isp_5.15.bb b/recipes-kernel/isp/isp_5.15.bb
index ca057322..4cfca805 100644
--- a/recipes-kernel/isp/isp_5.15.bb
+++ b/recipes-kernel/isp/isp_5.15.bb
@@ -23,6 +23,9 @@ do_compile() {
         oe_runmake -C ${S}/linux_54_t7c/ispHal/media-v4l2 M=${S}/linux_54_t7c/ispHal/media-v4l2 ${EXTRA_OEMAKE}
         oe_runmake -C ${S}/linux_54_t7c/ispHal/sensor_calibration M=${S}/linux_54_t7c/ispHal/sensor_calibration ${EXTRA_OEMAKE}
         oe_runmake -C ${S}/linux_54_t7c/ispHal/v4l2_test_media M=${S}/linux_54_t7c/ispHal/v4l2_test_media ${EXTRA_OEMAKE}
+    fi
+
+    if [ "${@bb.utils.contains("DISTRO_FEATURES", "arm-isp", "yes", "no", d)}" = "yes"  ]; then
         oe_runmake -C ${S}/linux_515_g12b/v4l2_dev M=${S}/linux_515_g12b/v4l2_dev ${EXTRA_OEMAKE}
         oe_runmake -C ${S}/linux_515_g12b/subdev/iq M=${S}/linux_515_g12b/subdev/iq ${EXTRA_OEMAKE}
         oe_runmake -C ${S}/linux_515_g12b/subdev/lens M=${S}/linux_515_g12b/subdev/lens ${EXTRA_OEMAKE}
@@ -52,6 +55,9 @@ do_install() {
         install -m 0666 ${S}/linux_54_t7c/ispHal/sensor_calibration/libtuning.so ${D}${libdir}
         install -m 0666 ${S}/linux_54_t7c/ispHal/include/*.h ${D}${includedir}
         install -m 0666 ${S}/linux_54_t7c/ispHal/sensor_calibration/imx290/*.h ${D}${includedir}
+    fi
+
+    if [ "${@bb.utils.contains("DISTRO_FEATURES", "arm-isp", "yes", "no", d)}" = "yes"  ]; then
         install -m 0666 ${S}/linux_515_g12b/v4l2_dev/iv009_isp.ko ${ISPDIR}
         install -m 0666 ${S}/linux_515_g12b/subdev/iq/iv009_isp_iq.ko ${ISPDIR}
         install -m 0666 ${S}/linux_515_g12b/subdev/lens/iv009_isp_lens.ko ${ISPDIR}
-- 
2.25.1

